lib/streamlit/delta_generator.py

```diff
@@ -879,4 +879,7 @@ def _enqueue_message(msg: ForwardMsg_pb2.ForwardMsg) -> None:
     if ctx is None:
         raise NoSessionContext()
 
+    if ctx.current_fragment_id and msg.WhichOneof("type") == "delta":
+        msg.delta.fragment_id = ctx.current_fragment_id
```

Comment:
question: the reason why this is in the `delta` message and not the `new_element` one is probably because blocks might also require deactivating `stale` effect, or?

Reply:
yep -- expanders and tabs in particular need to be marked as stale, so it made sense to add this to `